OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -50,10 +79,30 @@ public override bool RequestTransportNotify(Actor self, Actor carryable, CPos de
 			return false;
 		}
 
+		public override bool ReserveCarryable(Actor self, Actor carryable)
+		{
+			if (State == CarryallState.Reserved)
+				UnreserveCarryable(self);
+
+			var act = carryable.TraitOrDefault<AutoCarryable>();
+
+			if (act != null)
+			{
+				if (State != CarryallState.Idle || !act.AutoReserve(self, underAutoCommand))
+					return false;
+			}
+			else if (State != CarryallState.Idle || !carryable.Trait<Carryable>().Reserve(self))
```

Comment:
Just extract `State != CarryallState.Idle` from both of these clauses and place it before the AutoCarryable trait search

Reply:
should be fix